spec/Candid.md

```diff
@@ -740,8 +966,14 @@ variant { <fieldtype>;* } <: variant { <fieldtype'>;* } ~> f2
 ------------------------------------------------------------------------------------------------
 variant { <nat> : <datatype>; <fieldtype>;* } <: variant { <nat> : <datatype'>; <fieldtype'>;* }
   ~> \x.case x of <nat> y => <nat> (f1 y) | _ => f2 x
+
+opt variant { <fieldtype>;* } <: opt variant { <fieldtype'>;* } ~> f
+----------------------------------------------------------------------------------------------
+opt variant { <nat> : opt <datatype>; <fieldtype>;* } <: opt variant { <fieldtype'>;* }
+  ~> \x.join_opt (map_opt (\y.case y of <nat> z => null | _ => ?(f x)))
```

Comment:
This overlaps with the more general `opt` rule. It’s clear that it morally should have precendece, but still, is that a problem?